### PR TITLE
WIP: Be defensive against cloud plugins that don't report what we expect.

### PR DESCRIPTION
**Pull request by stevendanna**
Looking at the Ohai source code, it appears that cloud_v2 might be the
more consistent interface to use; however, in either case, I'd like to
do some more research to be sure.

Signed-off-by: Steven Danna <steve@chef.io>

### CODE REVIEW

**Comment by tas50** [comment]
Just a heads up if we move to Chef 13 the improvements to Ohai 13 help us out a bit here.

**Comment by stevendanna** [comment]
@tas50 Yeah, I think we never would have had this bug in Chef 13.

File: omnibus/files/private-chef-cookbooks/private-chef/libraries/nginx_erb.rb [hunk 1]
```diff
@@ -57,6 +55,29 @@ def ipv6?
     node['private_chef']['nginx']['enable_ipv6']
   end
 
+  def ip_or_nil
```
Comment (srenatus) [inline omnibus/files/private-chef-cookbooks/private-chef/libraries/nginx_erb.rb]: Isn't this missing an `(a)`? Or am I missing something?
```ruby
def ip_or_nil(a)
```

**Comment by markan** [comment]
Now that we've updated Chef and Ohai to 13 is this still needed?

**Comment by tas50** [comment]
This shouldn’t be an issue with modern Ohai.

**Comment by markan** [comment]
Closing because I think it's fixed by the ohai 13 upgrade. Feel free to reopen if I'm wrong.

**Pull request closed without merging**